searx/resources.py

```diff
@@ -0,0 +1,886 @@
+      name
+    - get properties like *EXISTS*
+
+    .. code-block:: python
+
+      >>> from searx.resources import File
+      >>> tmp = File('~/tmp') / 'foo'
+      >>> tmp
+      '/home/user/tmp/foo'
+      >>> tmp.EXISTS
+      False
+
+    No additional import, no juggling with ``os.join(...)``, simply slash ``/``
+    and ``foo.<method>`` calls
+
+    .. code-block:: python
+
+      >>> [(tmp/x).makedirs() for x in ('foo', 'bar')]
+      True, True
+      >>> for n in tmp.listdir():
+      ...     print(tmp / n)
+      ...
+      /home/user/tmp/foo
+      /home/user/tmp/bar
+
+    """
+
+    class OS:
+        # pylint: disable=too-few-public-methods, missing-class-docstring
+        sep     = os.sep
+        curdir  = os.curdir
+        altsep  = os.altsep
+        extsep  = os.extsep
+        pathsep = os.pathsep
+        defpath = os.defpath
+        linesep = os.linesep
+        devnull = os.devnull
+
+    OS = OS() # pylint: disable=invalid-name
+
+    def __new__(cls, pathname):
+        """Constructor of a path name object.
+
+        Regardless of how the encoding of the file system is, the ``pathname``
+        is converted to unicode.  The conversion of byte strings is based the
+        default encoding.
+
+        To issue "File-system Encoding" See also:
+
+        * https://docs.python.org/3/howto/unicode.html#unicode-filenames
+
+        """
+        pathname = path.normpath(path.expanduser(str(pathname)))
+        return super(File, cls).__new__(cls, pathname)
+
+    @property
+    def VALUE(self):
+        """string of the path name"""
+        return str(self)
+
+    @property
+    def EXISTS(self):
+        """True if file/pathname exist (:py:func:`os.path.exists`)"""
+        return path.exists(self)
+
+    @property
+    def SIZE(self):
+        """Size in bytes (:py:func:`os.path.getsize`)"""
+        return path.getsize(self)
+
+    @property
+    def READABLE(self):
+        """True if file/path is readable (:py:func:`os.access`)"""
+        return os.access(self, os.R_OK)
+
+    @property
+    def WRITEABLE(self):
+        """True if file/path is writeable (:py:func:`os.access`)"""
+        return os.access(self, os.W_OK)
+
+    @property
+    def EXECUTABLE(self):
+        """True if file is executable (:py:func:`os.access`)"""
+        return os.access(self, os.X_OK)
+
+    @property
+    def ISDIR(self):
+        """True if path is a folder (:py:func:`os.path.isdir`)"""
+        return path.isdir(self)
+
+    @property
+    def ISFILE(self):
+        """True if path is a file (:py:func:`os.path.isfile`)"""
+        return path.isfile(self)
+
+    @property
+    def ISABSPATH(self):
+        """True if path is absolute (:py:func:`os.path.isabs`)"""
+        return path.isabs(self)
+
+    @property
+    def ISLINK(self):
+        """True if path is a symbolic link (:py:func:`os.path.islink`)"""
+        return path.islink(self)
+
+    @property
+    def ISMOUNT(self):
```

Comment:
These are properties ... 

````python
if myfile.EXISTS:
     print("file size: %s" % (myfile.SIZE))
````

I decided to use uppercase since this is in tradition with [os.stat](https://docs.python.org/3/library/stat.html).

> As I see, only a few of them are actually used in the current code.

It is a lib we can make use of. It helps me (and I think others also) to handle file resources more expressive and readable (as I mentioned near to the other remarks from you).

Reply:
My note isn't about the uppercasing, it's about the so many unused functions what doesn't add any value to the current codebase. I consider these functions as [dead code](https://en.wikipedia.org/wiki/Dead_code) which can lead to numerous issues in the future and it also makes reviewing slower and harder. Could you eliminate unused functions?

Reply:
>  it's about the so many unused functions

They exists for programmers convenience, to implement more readable source code. If aspects like [dead code](https://en.wikipedia.org/wiki/Dead_code) or [unreachable code](https://en.wikipedia.org/wiki/Unreachable_code) overrule _comfort_ we should remove it.   But isn't it the same with every lib we use? If _unused_ names are a only KO criteria for the merge, I can clean up the code.

Reply:
Just an idea from mine, I don't know if it fits to your request: .. I tested all functions, when I implemented them, I can also implement unit tests ..